### PR TITLE
Fix Schema Editor not handling empty states

### PR DESCRIPTION
**Pull request by agalin920**
Refactor schema structure to match content editor and add loading states to properly handle empty states

https://www.loom.com/share/4ccbdefb489a4e03b3fe6c256e76296d

### CODE REVIEW

File: src/apps/schema/src/app/main.js [hunk 1]
```diff
@@ -1,30 +1,44 @@
...
   const match = useRouteMatch("/schema/start");
   const showNav = match && match.isExact;
+  const navSchema = useSelector((state) => state.navSchema);
+  const dispatch = useDispatch();
+
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    setLoading(true);
+    Promise.all([dispatch(fetchModels()), dispatch(fetchSettings())]).then(
```
Comment (shrunyan) [inline src/apps/schema/src/app/main.js:25]: Should we include a `catch` clause to handle if an API request fails?

Comment (agalin920) [reply src/apps/schema/src/app/main.js:25]: Sure. What is our standard to handle API failures? Can you point me to an existing case to base myself off on? Should we also add some to the content-editor .then chain?

Comment (shrunyan) [reply src/apps/schema/src/app/main.js]: This is probably a good example: https://github.com/zesty-io/manager-ui/blob/master/src/apps/audit-trail/src/views/app/index.js#L26

In an api request failure we'd want to stop the loading state and show an error notification

Comment (agalin920) [reply src/apps/schema/src/app/main.js]: Sounds good. On it

**Review by shrunyan — APPROVED** [review]

**Pull request merged**